app/controllers/concerns/phone_confirmation_flow.rb

```diff
@@ -65,7 +72,26 @@ def send_confirmation_code
     # user's session. Re-sending the confirmation code doesn't generate a new one.
     self.confirmation_code = generate_confirmation_code unless confirmation_code
 
-    SmsSenderOtpJob.perform_later(confirmation_code, unconfirmed_phone)
+    if current_otp_delivery_method == :voice
```

Comment:
The reason is for fallback.. if user mucks with query parameter, `current_otp_delivery_method` will be `nil` and sender will use SMS.